services/ui_backend_service/api/notify.py

```diff
@@ -34,16 +33,23 @@ def __init__(self, app, db, event_emitter=None):
         self.loop.create_task(self._init(self.db.pool))
 
     async def _init(self, pool):
-        async with pool.acquire() as conn:
-            await asyncio.gather(
-                self.listen(conn),
-                self.ping(conn)
-            )
+        while True:
+            try:
+                async with pool.acquire() as conn:
```

Comment:
can this cause some unexpected effects? the `while` / `async with` is in a different order compared to the previous implementation

Reply:
I'm not completely sure what you mean, there wasn't a `while` loop in the previous implementation. Once coroutines(listen/ping) are done execution leaves the context manager and previous connection is closed before new connection is acquired from the pool.